File: discover_ruby/github_most_popular.rb
require 'httpclient'

def req(extheaders, query)
	client = HTTPClient.new
	client.get_content('https://api.github.com/search/repositories?', query, extheaders)
end

extheaders = {
  'User-Agent' => 'Holberton_School',
  'Authorization' => 'token fa51959c07bf48159fbaa3c6800b12efb49f217e'
}

query=["q", "language:ruby"], ["sort", "stars"], ["order", "desc"]

puts req(extheaders, query)

File: discover_more_python/github_most_popular.py
import urllib2

def http_get(url, headers):
	req = urllib2.Request(url, None, headers)
	response = urllib2.urlopen(req)
	return response.read()
	
request_headers = { 'User-Agent': 'Holberton_School', 'Authorization': 'token fa51959c07bf48159fbaa3c6800b12efb49f217e' }
print http_get('https://api.github.com/search/repositories?q=language:python&sort=stars&order=desc', request_headers)
